### PR TITLE
Call train test with default config

### PR DESCRIPTION
**Pull request by FreddyMurphy**
If there hasn't been created any validation set ml tooling will now
create one using default configurations

- [x] close issue #439
- [x] Tests added
- [x] Passes flake8
- [x] Updated CHANGELOG

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # v0.10.4
 - Added `load_demo_dataset` function
+- If the dataset has no train set `score_estimor` will now run `create_train_test` with default configurations
```
Comment (andersbogsnes) [inline CHANGELOG.md:3]:
```suggestion
- If the dataset has no train set `score_estimator` will now run `create_train_test` with default configurations
```

File: src/ml_tooling/baseclass.py [hunk 2]
```diff
@@ -408,7 +409,10 @@ def score_estimator(
         logger.info("Scoring estimator...")
 
         if not data.has_validation_set:
-            raise DatasetError("Must run create_train_test first!")
+            data.create_train_test(
+                stratify=True if self.is_classifier else False,
+                seed=self.config.RANDOM_STATE,
```
Comment (andersbogsnes) [inline src/ml_tooling/baseclass.py:414]: We should add the other configuration options here as as well.

File: src/ml_tooling/baseclass.py [hunk 3]
```diff
@@ -408,7 +409,10 @@ def score_estimator(
         logger.info("Scoring estimator...")
 
         if not data.has_validation_set:
-            raise DatasetError("Must run create_train_test first!")
+            data.create_train_test(
+                stratify=True if self.is_classifier else False,
```
Comment (andersbogsnes) [inline src/ml_tooling/baseclass.py:413]: `self.is_classifier` returns True or False, no need for the `if`

File: tests/test_baseclass.py [hunk 4]
```diff
@@ -431,12 +431,40 @@ def load_prediction_data(self, *args, **kwargs):
 
 
 class TestScoreEstimator:
-    def test_score_estimator_fails_if_no_train_test_data_available(self, iris_dataset):
+    def test_score_estimator_if_no_train_test_data_available(
```
Comment (andersbogsnes) [inline tests/test_baseclass.py:434]: What is this test doing? Hard to understand from the function name

File: tests/test_baseclass.py [hunk 5]
```diff
@@ -431,12 +431,40 @@ def load_prediction_data(self, *args, **kwargs):
...
 
-        with pytest.raises(MLToolingError, match="Must run create_train_test first!"):
-            model.score_estimator(data)
+        test = train_iris_dataset
+        result = model.score_estimator(test)
+
+        assert model.score_estimator(data) == result
+
+    def test_score_estimator_if_no_train_test_data_available_changed_config(
+        self, iris_dataset, train_iris_dataset
```
Comment (andersbogsnes) [inline tests/test_baseclass.py:457]: Looks like `train_iris_dataset` is not being used?

What is this testing?

File: tests/test_baseclass.py (hunk 4 above)
Comment (FreddyMurphy) [reply tests/test_baseclass.py:434]: This test checks if the `score_estimator ` correctly calculates the score estimator without any test data has been created
```suggestion
    def test_score_estimator_creates_train_test_data(
```

File: tests/test_baseclass.py (hunk 5 above)
Comment (FreddyMurphy) [reply tests/test_baseclass.py:457]: `train_iris_dataset` should not be included and has been removed.

This test checks  if the `score_estimator` correctly calculates the score estimator without any test data has been create and the config has been changed
```suggestion
        def test_score_estimator_creates_train_test_data_with_changed_config(
        self, iris_dataset
```

File: src/ml_tooling/baseclass.py (hunk 2 above)
Comment (FreddyMurphy) [reply src/ml_tooling/baseclass.py:416]: They have been added in the newest commit

File: src/ml_tooling/baseclass.py (hunk 3 above)
Comment (FreddyMurphy) [reply src/ml_tooling/baseclass.py]: The change has been implemented in the newest commit

File: tests/test_baseclass.py [hunk 6]
```diff
@@ -431,12 +431,59 @@ def load_prediction_data(self, *args, **kwargs):
...
+        model = Model(LogisticRegression())
+        model.config.RANDOM_STATE = 1
+        model.config.TEST_SIZE = 0.50
+        data = iris_dataset()
+
+        test = iris_dataset()
+        test.create_train_test(stratify=True, seed=1, test_size=0.50)
+        result = model.score_estimator(test)
+
+        assert model.score_estimator(data) == result
```
Comment (andersbogsnes) [inline tests/test_baseclass.py:485]: I would probably just compare the two datasets - do they have the same x and y?

**Pull request merged**